meos/src/general/ttext_textfuncs_meos.c

```diff
@@ -41,6 +41,21 @@
  * Text concatenation
  *****************************************************************************/
 
+/**
+ * @ingroup meos_temporal_text
+ * @brief Return the concatenation of two texts
+ * @param[in] txt1,txt2 Values
+ */
+text *
+textcat_text_text(const text *txt1, const text *txt2)
+{
+  /* Ensure validity of the arguments */
+  if (! ensure_not_null((void *) txt1) || ! ensure_not_null((void *) txt2))
+    return NULL;
+  return DatumGetTextP(datum_textcat(PointerGetDatum(txt1), 
+    PointerGetDatum(txt2)));
```

Comment:
I believe we have a function called `text_catenate`, which uses `text *`. Here we are casting to `Datum` simply for `datum_textcat` to cast back to `text *` and call `text_catenate`